### PR TITLE
add testutils for Apache Kafka (GDEV-1210)

### PR DESCRIPTION
**Pull request by KerstenBreuer**
Description:
```
Added testutils including a pytest fixture an event recorder that
can be used to check for published events.

Bump version to 0.4.0.
```

### CODE REVIEW

File: hexkit/providers/akafka/provider.py [hunk 1]
```diff
@@ -83,17 +83,19 @@ class KafkaProducerCompatible(Protocol):
...
         Initialize the producer with some config params.
         Args:
-            bootstrap_servers (list[str]):
-                List of connection strings pointing to the kafka brokers.
-            client_id (str):
+            bootstrap_servers:
+                Comma-separated list of connection strings pointing to the kafka
+                brokers.
+            client_id:
                 A globally unique ID identifying this the Kafka client.
```
Comment (Cito) [inline hexkit/providers/akafka/provider.py:99]:
```suggestion
                A globally unique ID identifying the Kafka client.
```

File: hexkit/providers/akafka/provider.py [hunk 2]
```diff
@@ -213,9 +226,10 @@ def __init__(
...
 
         Args:
-            bootstrap_servers (list[str]):
-                List of connection strings pointing to the kafka brokers.
-            client_id (str):
+            bootstrap_servers:
+                Comma-separated list of connection strings pointing to the kafka
+                brokers.
+            client_id:
                 A globally unique ID identifying this the Kafka client.
```
Comment (Cito) [inline hexkit/providers/akafka/provider.py:233]:
```suggestion
                A globally unique ID identifying the Kafka client.
```

File: hexkit/providers/akafka/provider.py [hunk 3]
```diff
@@ -196,13 +198,24 @@ class ConsumerEvent(Protocol):
...
+
+
 class KafkaConsumerCompatible(Protocol):
     """A python duck type protocol describing an AIOKafkaConsumer or equivalent."""
 
     def __init__(
         self,
-        *topics,
-        bootstrap_servers: list[str],
+        *topics: str,
```
Comment (Cito) [inline hexkit/providers/akafka/provider.py:217]: Shouldn't this be Ascii?

File: hexkit/providers/akafka/testutils.py [hunk 4]
```diff
@@ -0,0 +1,305 @@
...
+    def __init__(self, kafka_servers: list[str], publisher: KafkaEventPublisher):
+        """Initialize with connection details and a ready-to-use publisher."""
+
+        self.kafka_servers = kafka_servers
+        self.publisher = publisher
+
+    async def publish_event(
+        self, *, payload: JsonObject, type_: Ascii, topic: Ascii, key: Ascii = "test"
+    ) -> None:
+        """A convienience method to publish a test event."""
```
Comment (Cito) [inline hexkit/providers/akafka/testutils.py:271]:
```suggestion
        """A convenience method to publish a test event."""
```

File: tests/fixtures/utils.py [hunk 5]
```diff
@@ -16,9 +16,7 @@
 """Uitls for Fixture handling"""
```
Comment (Cito) [inline tests/fixtures/utils.py:16]:
```suggestion
"""Utils for fixture handling"""
```

File: tests/integration/test_akafka_testutils.py [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+            "test_key",
+        ),
+    ],
+)
+async def test_event_recorder_missmatch(
+    events_to_publish: Sequence[ExpectedEvent],
+    key_to_publish: str,
+    kafka_fixture: KafkaFixture,  # noqa: F811
+):
+    """Test the handling of missmatches between recorded and expected events.
```
Comment (Cito) [inline tests/integration/test_akafka_testutils.py:94]:
```suggestion
    """Test the handling of mismatches between recorded and expected events.
```

**Review by Cito — APPROVED** [review]
Super nice.

File: hexkit/providers/akafka/provider.py (hunk 3 above)
Comment (KerstenBreuer) [reply hexkit/providers/akafka/provider.py:217]: Right.

**Pull request merged**